services/provider/server/server.go

```diff
@@ -655,14 +654,6 @@ func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req
 func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStatusRequest) (*pb.ReportStatusResponse, error) {
 	// Update the status in storageConsumer CR
 
-	if _, err := semver.Parse(req.ClientPlatformVersion); err != nil {
```

Comment:
This `ReportStatus` is called and executed at client end, do you mean we need to have validation at Client as well?

Reply:
AFAIK `ReportStatus` is the RPC implementation on the server side and this validation is also required here only if the values are set.

Reply:
Here `ReportStatus` is an abstraction over actual gRPC, what we are trying to achieve is to have the data consumer validate and use/discard the data.

The reason for above is, updation of timestamp on storageconsumer CR is special when compared to these values and we don't want timestamp to get effected at the end.

Reply:
ack, you want client side validation as well, in that scenario we want to take that as part of different PR.

Then the requirement will be, fail as early as possible and also update CR that we heard from client even if validation of other fields fail.

Is that the expectation?

Reply:
You already have the client validation isnt it? For me its missing on the server side.

Reply:
sorry i missed it, even though we dont have a client-side, am not sure about the need for it.

Reply:
- this PR is moving client side validation to server side usage site
- the intention is, validation shouldn't inhibit us from updating heartbeat
- this PR doesn't intend to differentiate b/n how data and heartbeat should be handled, like, should we branch if data is malformed and update only heartbeat?

Reply:
I feel that invalid data is handled but logged as a warning and getting continued, it should be rejected by the server.

Reply:
why do we need to consider serving part of the request when the user sends a request with invalid data?, i would suggest sending an invalid error response and don't update the CR and the user needs to take care of setting all the parameters when sending the request. @nb-ohad thoughts?

Reply:
>Reintroduce the validation to the server side, but allow empty strings as valid values
👍🏻 
>Validation failures should be recorded in the log, but should not stop the processing of the request
If this is on the Client side am good, but if we are processing the request incase of validation failure is not correct on the server side.

>Accepting empty values should be allowed, but should have a warning (not an error) in the log to indicate that we got partial data.

Even we don't need to log a warning, this should be logged as debug/info because the parameter is optional in the Request.

> When moving down the processing pipeline (we should replace empty values with the string "N/A"). For example when setting values on the CRs status

Reply:
From above I'm drawing these two differences.
Ohad: If validation fails, the default value will be "N/A" and log the validation failure but also update the hearbeat
Madhu: If validation fails, return the error w/o updating the heartbeat

The arguments align around empty values (not exact but similar), so no issue there.

I agree w/ Madhu on a Request-Response cycle and at the same time, here we don't have generalized situation. So, I don't have any decision to lean on.

---

This is what I'm willing to, introduce the validation removal and removing validation of empty values for now.

Reply:
done.